ee/app/engagement-dashboard/client/components/UsersTab/ActiveUsersSection.js

```diff
@@ -47,37 +46,38 @@ const ActiveUsersSection = () => {
 
 		const createPoints = () => Array.from({ length: moment(period.end).diff(period.start, 'days') + 1 }, (_, i) => createPoint(i));
 
-		const distributeValueOverPoints = (value, i, T, array, prev) => {
-			for (let j = 0; j < T; ++j) {
-				const k = i + j;
-
-				if (k >= array.length) {
-					continue;
-				}
-
-				if (k >= 0) {
-					array[k].y += value;
-				}
-
-				if (k === -1) {
-					prev.y += value;
-				}
-			}
-		};
-
 		const dauValues = createPoints();
 		const prevDauValue = createPoint(-1);
 		const wauValues = createPoints();
 		const prevWauValue = createPoint(-1);
 		const mauValues = createPoints();
 		const prevMauValue = createPoint(-1);
 
-		for (const { users, day, month, year } of data.month) {
-			const i = moment.utc([year, month - 1, day, 0, 0, 0, 0]).diff(period.start, 'days');
-			distributeValueOverPoints(users, i, 1, dauValues, prevDauValue);
-			distributeValueOverPoints(users, i, 7, wauValues, prevWauValue);
-			distributeValueOverPoints(users, i, 30, mauValues, prevMauValue);
+		const usersListsMap = data.month.reduce((map, dayData) => {
+			const date = moment({ year: dayData.year, month: dayData.month - 1, day: dayData.day });
+			const i = date.diff(period.start, 'days');
```

Comment:
what's `i`? 👀

Reply:
@KevLehman I'm not sure how to properly name a "date offset over a time inverval, in days" variable. Suggestions?

Reply:
I'd go with `dateOffset` or just `day`, since it represents the day to which `dayData` belongs in the period, isn't it? 👀

Reply:
Good suggestion, I've changed `i` to `dateOffset` 🙂